src/braket/circuits/circuit.py

```diff
@@ -194,7 +198,7 @@ def add_result_type(
         self,
         result_type: ResultType,
         target: QubitSetInput = None,
-        target_mapping: Dict[QubitInput, QubitInput] = {},
+        target_mapping: Dict[QubitInput, QubitInput] = None,
```

Comment:
Why?

Reply:
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/#:~:text=In%20Python%2C%20when%20passing%20a,or%20even%20a%20class%20instance.

Will never forget the headaches this has caused me in the past